specification/purview/resource-manager/Microsoft.Purview/preview/2023-05-01-preview/purview.json

```diff
@@ -0,0 +1,2980 @@
+          },
+          {
+            "in": "body",
+            "name": "featureRequest",
+            "description": "Request body with feature names.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BatchFeatureRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/BatchFeatureStatus"
+            }
+          },
+          "default": {
+            "description": "An error response received from the provider.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Features_SubscriptionGet": {
+            "$ref": "./examples/Features_SubscriptionGet.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Purview/accounts/{accountName}/listFeatures": {
+      "post": {
+        "tags": [
+          "Feature"
+        ],
+        "summary": "Gets a list of features and their status for the account.\r\nStatus of enabled features will be true. Status of disabled features will be false.\r\nFeatures that don't exist will be excluded from the results.",
+        "description": "Gets details from a list of feature names.",
+        "operationId": "Features_AccountGet",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/accountName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "in": "body",
+            "name": "featureRequest",
+            "description": "Request body with feature names.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BatchFeatureRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/BatchFeatureStatus"
+            }
+          },
+          "default": {
+            "description": "An error response received from the provider.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Features_AccountGet": {
+            "$ref": "./examples/Features_AccountGet.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Purview/accounts/{accountName}/ingestionPrivateEndpointConnections": {
```

Comment:
>ingestionPrivateEndpointConnections [](http://example.com/codeflow?start=123&length=35)

do you not have individual gets for the resources ? I only see the collection GET operation here

Reply:
That's correct. We do not have individual GET operations for these resources.

Reply:
I mean why not ? The customers\arm tools  expect that each resource will be individually gettable.

Reply:
Same as the [response below](https://github.com/Azure/azure-rest-api-specs/pull/25844#discussion_r1339384883).

Also, Purview UX portal only provides LIST and POST operation on the `ingestionPrivateEndpointConnection`; it has no view for GET individual resources. So, a GET API is not supported or required.

Reply:
No this is not really acceptable. When you return a list with the resources , Tools written on ARM and ARM customers will expect to able to do a GET on the id property for individual resources that is returned here . Please expose individual GETs on these resoruces as well

![image](https://github.com/Azure/azure-rest-api-specs/assets/11067581/0eb08dac-3ddb-4b8e-976d-d4565c8b6ec4)

Reply:
Hi Suhas,

Thanks for the suggestions. The consumers of these preview APIs for Purview are internal UX services.  We agree that it would have been good to have these as Proxy resources, but it will entail a lot of rework at the moment for Purview who are targeting to use these preview API versions for Microsoft Ignite.

Based on customer feedback, we will be able to asses and add proxy resource and individual gets when we finalize the preview versions. Please let us know if this helps.

Thanks,
Amol Dixit

Reply:
Approving this since this is mainly internal apis using the service